### ISSUE TITLE
Hibernate 6.3.1 breaking changes

### ISSUE THREAD

**Issue by ver-stefan** [liquibase/liquibase-hibernate#557]
liquibase-hibernate6 4.23.2 doesn't seem to support the latest hibernate version 6.3.1.Final fully.

This code:
```java
String package = "com.mycompany.myapp.entity";
String dialect= "org.hibernate.dialect.MySQLDialect";
String url="hibernate:spring:" + package + "?dialect=" + dialect;
Connection hibernateConnection = new HibernateConnection(url, new ClassLoaderResourceAccessor());
Database hibernate = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(new JdbcConnection(hibernateConnection));
```

Results in:
```
java.lang.NoSuchMethodError: 'java.util.Iterator org.hibernate.mapping.Value.getColumnIterator()'

	at org.hibernate.envers.boot.model.ColumnContainer.addColumnsFromValue(ColumnContainer.java:41)
	at org.hibernate.envers.configuration.internal.metadata.BasicMetadataGenerator.buildProperty(BasicMetadataGenerator.java:100)
	at org.hibernate.envers.configuration.internal.metadata.BasicMetadataGenerator.addBasic(BasicMetadataGenerator.java:47)
	at org.hibernate.envers.configuration.internal.metadata.IdMetadataGenerator.addBasic(IdMetadataGenerator.java:307)
	at org.hibernate.envers.configuration.internal.metadata.IdMetadataGenerator.addIdProperty(IdMetadataGenerator.java:112)
	at org.hibernate.envers.configuration.internal.metadata.IdMetadataGenerator.addIdProperties(IdMetadataGenerator.java:126)
	at org.hibernate.envers.configuration.internal.metadata.IdMetadataGenerator.addIdProperty(IdMetadataGenerator.java:109)
	at org.hibernate.envers.configuration.internal.metadata.IdMetadataGenerator.addIdProperties(IdMetadataGenerator.java:126)
	at org.hibernate.envers.configuration.internal.metadata.IdMetadataGenerator.addIdAndGetMappingData(IdMetadataGenerator.java:221)
	at org.hibernate.envers.configuration.internal.metadata.AuditMetadataGenerator.generateFirstPass(AuditMetadataGenerator.java:270)
	at org.hibernate.envers.configuration.internal.EntitiesConfigurator.configure(EntitiesConfigurator.java:77)
	at org.hibernate.envers.boot.internal.EnversServiceImpl.initialize(EnversServiceImpl.java:129)
	at org.hibernate.envers.boot.internal.AdditionalMappingContributorImpl.contribute(AdditionalMappingContributorImpl.java:50)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.lambda$processAdditionalMappingContributions$0(MetadataBuildingProcess.java:383)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1092)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.processAdditionalMappingContributions(MetadataBuildingProcess.java:380)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:338)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:1432)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1503)
	at liquibase.ext.hibernate.database.HibernateEjb3Database.buildMetadataFromPath(HibernateEjb3Database.java:56)
	at liquibase.ext.hibernate.database.HibernateDatabase.buildMetadata(HibernateDatabase.java:133)
	at liquibase.ext.hibernate.database.HibernateDatabase.setConnection(HibernateDatabase.java:73)
	at liquibase.database.DatabaseFactory.findCorrectDatabaseImplementation(DatabaseFactory.java:124)
```

Will this be fixed this in 4.23.3, and if so, is there a planned release date?

**Comment by ver-stefan** [comment]
I might have an environment with mixed Hibernate versions.
Disregard this for now.....

**Comment by ver-stefan** [comment]
The above exception was related to mixed versions between hibernate-core and hibernate-envers and was resolved by using the same version 6.3.1.Final for both.

However I get a similar exception later when comparing two databases, which doesn't seem to be related to mixed hibernate version:

```
java.lang.NoSuchMethodError: 'java.util.Iterator org.hibernate.mapping.Table.getColumnIterator()'

	at liquibase.ext.hibernate.snapshot.UniqueConstraintSnapshotGenerator.addTo(UniqueConstraintSnapshotGenerator.java:65)
	at liquibase.ext.hibernate.snapshot.HibernateSnapshotGenerator.snapshot(HibernateSnapshotGenerator.java:82)
	at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
	at liquibase.ext.hibernate.snapshot.HibernateSnapshotGenerator.snapshot(HibernateSnapshotGenerator.java:74)
	at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
	at liquibase.ext.hibernate.snapshot.HibernateSnapshotGenerator.snapshot(HibernateSnapshotGenerator.java:74)
	at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
	at liquibase.ext.hibernate.snapshot.HibernateSnapshotGenerator.snapshot(HibernateSnapshotGenerator.java:74)
	at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
	at liquibase.ext.hibernate.snapshot.HibernateSnapshotGenerator.snapshot(HibernateSnapshotGenerator.java:74)
	at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49)
	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:313)
	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:456)
	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:478)
	at liquibase.snapshot.DatabaseSnapshot.includeNestedObjects(DatabaseSnapshot.java:369)
	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:332)
	at liquibase.snapshot.DatabaseSnapshot.init(DatabaseSnapshot.java:106)
	at liquibase.snapshot.DatabaseSnapshot.<init>(DatabaseSnapshot.java:59)
	at liquibase.snapshot.JdbcDatabaseSnapshot.<init>(JdbcDatabaseSnapshot.java:34)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:214)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:187)
	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:154)
	at liquibase.diff.DiffGeneratorFactory.compare(DiffGeneratorFactory.java:67)
	at liquibase.diff.DiffGeneratorFactory.compare(DiffGeneratorFactory.java:63)
```

So the question still stands:
Is this a known issue that will be fixed this in 4.23.3, and if so, is there a planned release date?

**Comment by ver-stefan** [comment]
The issue disappeared with version 4.24.0.
Thanks!
